From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 14 Aug 2017 09:47:31 +0200
Subject: Revert "/proc/iomem: only expose physical resource addresses to
 privileged users"
Patch-mainline: never, kabi
References: kabi

This reverts commit e8aff60373182f48f5191b147894e954a591a521, upstream
commit 51d7b120418e99d6b3bf8df9eb3cc31e8171dee4. It relies on commit
14ae9c4b5ae235bdaa74bb1c40de9a42ca691566, upstream commit
34dbbcdbf63360661ff7bda6c5f52f99ac515f92 which was reverted due to kABI
breakage.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/resource.c |   13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

--- a/kernel/resource.c
+++ b/kernel/resource.c
@@ -105,25 +105,16 @@ static int r_show(struct seq_file *m, vo
 {
 	struct resource *root = m->private;
 	struct resource *r = v, *p;
-	unsigned long long start, end;
 	int width = root->end < 0x10000 ? 4 : 8;
 	int depth;
 
 	for (depth = 0, p = r; depth < MAX_IORES_LEVEL; depth++, p = p->parent)
 		if (p->parent == root)
 			break;
-
-	if (file_ns_capable(m->file, &init_user_ns, CAP_SYS_ADMIN)) {
-		start = r->start;
-		end = r->end;
-	} else {
-		start = end = 0;
-	}
-
 	seq_printf(m, "%*s%0*llx-%0*llx : %s\n",
 			depth * 2, "",
-			width, start,
-			width, end,
+			width, (unsigned long long) r->start,
+			width, (unsigned long long) r->end,
 			r->name ? r->name : "<BAD>");
 	return 0;
 }
